### PR TITLE
redis conf refactor

### PR DESCRIPTION
**Pull request by snopoke**
buddies @sravfeyn @esoergel 

Can review per commit :fish:

### CODE REVIEW

File: environments/production/inventory.ini [hunk 1]
```diff
@@ -71,7 +71,7 @@ hqtouch0.internal-va.commcarehq.org
 hqformplayer1.internal-va.commcarehq.org
 
 [redis]
-redis0.internal-va.commcarehq.org
+redis0.internal-va.commcarehq.org datavol_device=/dev/xvdb
```
Comment (snopoke) [inline environments/production/inventory.ini]: @NitigyaS FYI you forgot to specify this for the encrypted drive

**Comment by snopoke** [comment]
Deployed to `redis0.internal-va.commcarehq.org`

File: ansible/group_vars/all.yml [hunk 2]
```diff
@@ -166,11 +166,20 @@ supervisor_http_password: "{{ secrets.SUPERVISOR_HTTP_PASSWORD | default(None) }
...
 # this reads "'s3.{{ aws_region }}.amazonaws.com' if aws_region else None"
 aws_endpoint: '{{ aws_region and "s3." + aws_region + ".amazonaws.com" }}'
+
+redis_max_memory: 4096mb
+redis_logfile: '/var/log/redis/redis-server.log'
+redis_syslog_enabled: 'no'
+redis_database_save_times: []
+redis_max_clients: 10000
+redis_maxmemory_policy: allkeys-lru
+redis_appendonly: 'yes'
```
Comment (emord) [inline ansible/group_vars/all.yml]: @snopoke Does this have precedence over `ansible/roles/redis/defaults/main.yml` ? It looks like that defaults to `no`. it also seems like we're defining "defaults" in two places which seems like an anti-pattern

Comment (snopoke) [reply ansible/group_vars/all.yml]: Maybe overboard but https://github.com/dimagi/commcare-cloud/pull/1503

**Pull request merged**